wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/modal/ModalDialog.js

```diff
@@ -0,0 +1,424 @@
+			var body = $("body");
+			body.addClass("modal-dialog-open modal-dialog-no-scroll");
+
+			scrollTop = $(window).scrollTop();
+
+            if (!!navigator.platform && /iPad|iPhone|iPod/.test(navigator.platform)) {
+                body.addClass("modal-dialog-open-ios");
+            }
+		},
+		terminate : function() {
+			$("body").removeClass("modal-dialog-open modal-dialog-no-scroll modal-dialog-open-ios");
+            if (!!navigator.platform && /iPad|iPhone|iPod/.test(navigator.platform)) {
+            	$(window).scrollTop(scrollTop);
+            }
+		}
+	};
+
+	/**
+	 * Behavior that memorizes the focussed element when dialog is opened, and
+	 * returns focus to it when dialog is closed
+	 */
+	var returnFocusOnCloseBehavior = {
+		open : function(overlay, element, data) {
+			data.opener = document.activeElement;
+			if (data.options.validate) {
+				if (!data.opener || $(data.opener).is("body")) {
+					data.options.console.error("Error saving focused element when opening the modal, it is either none or body: ",
+							data.opener);
+				}
+			}
+		},
+		close : function(overlay, element, data) {
+			if (data.opener) {
+				try {
+					data.opener.focus();
+				} catch (error) {
+					if (data.options.validate) {
+						data.options.console.error(
+								"Error restoring focus after modal is closed. Attempted to set focus to element, but got an exception",
+								data.opener, error);
+					}
+					throw error;
+				}
+			}
+		}
+	}
+
+	/** Takes care of adding any necessary aria-related attributes to the dialog */
+	var addAriaAttributesBehavior = {
+		prepare : function(overlay, element, data) {
+			var content = overlay.find(CONTENT_SELECTOR);
+			var attrs = {
+				"role" : "dialog",
+				"aria-modal" : "true"
+			};
+
+			var title = element.find(".x-modal-title").first();
+			if (title.length > 0) {
+				attrs["aria-labelledby"] = getOrCreateId(title);
+			} else if (data.options.validate) {
+				data.options.console.error("No .x-modal-title element present in modal content: ", element.get(0));
+			}
+
+			var description = element.find(".x-modal-description").first();
+			if (description.length > 0) {
+				attrs["aria-describedby"] = getOrCreateId(description);
+			} else if (data.options.validate) {
+				data.options.console.warn("No .x-modal-description element present in modal content: ", element.get(0));
+			}
+
+			content.attr(attrs);
+		}
+	}
+
+	/** Closes the modal if the overlay is clicked or an escape key is pressed */
+	var closeOnOverlayClickOrEscapeBehavior = {
+		prepare : function(overlay, element, data) {
+			if (data.options.closeOnClickOutside) {
+				overlay.on("click.modal-dialog", function (event) {
+					if ($(event.target).closest(CONTENT_SELECTOR).length === 0) {
+						// clicked outside modal's content
+						findAndClickCloseButton(element);
+					}
+				});
+			}
+			if (data.options.closeOnEscape) {
+				overlay.on("keydown", function (event) {
+					if (event.which == 27) {
```

Comment:
JSHint (once ModalDialog.js is added to testing/wicket-js-tests/Gruntfile.js) will complain here on `==`. We should use `===` instead.